pcawg_metadata_parser/compare_xml_md5sum.py

```diff
@@ -0,0 +1,250 @@
+    xml_str = re.sub(r'<refassem_short_name>.+?</refassem_short_name>', '<refassem_short_name></refassem_short_name>', xml_str)
+    #xml_str = re.sub(r'<VALUE>.*{"qc_metrics".+?</VALUE>', '<VALUE>\n{"qc_metrics"}</VALUE>', xml_str, re.DOTALL)
+
+
+    xml_str = re.sub(r'<STUDY_REF .+?/>', '<STUDY_REF/>', xml_str)
+    xml_str = re.sub(r'<ANALYSIS_SET .+?>', '<ANALYSIS_SET>', xml_str)
+    xml_str = re.sub(r'<ANALYSIS .+?>', '<ANALYSIS>', xml_str)
+    xml_str = re.sub(r'<EXPERIMENT_SET .+?>', '<EXPERIMENT_SET>', xml_str)
+    xml_str = re.sub(r'<RUN_SET .+?>', '<RUN_SET>', xml_str)
+    xml_str = re.sub(r'<analysis_detail_uri>.+?</analysis_detail_uri>', '<analysis_detail_uri></analysis_detail_uri>', xml_str)
+    xml_str = re.sub(r'<analysis_submission_uri>.+?</analysis_submission_uri>', '<analysis_submission_uri></analysis_submission_uri>', xml_str)
+    xml_str = re.sub(r'<analysis_data_uri>.+?</analysis_data_uri>', '<analysis_data_uri></analysis_data_uri>', xml_str)
+
+    # we need to take care of xml properties in different order but effectively/semantically the same
+    gnos_analysis = xmltodict.parse(xml_str).get('ResultSet').get('Result')
+    effective_eq_xml = json.dumps(gnos_analysis, indent=4, sort_keys=True)
+    md5sum.append(hashlib.md5(effective_eq_xml).hexdigest())
+
+    # take out the qc_metrics section
+    analysis_attrib = get_analysis_attrib(gnos_analysis)
+    qc_metrics_xml = json.dumps(json.loads(analysis_attrib.get('qc_metrics')) if analysis_attrib.get('qc_metrics') else [], indent=4, sort_keys=True)
+    md5sum.append(hashlib.md5(qc_metrics_xml).hexdigest())
+
+    # take out the different file_type section
+    for file_type in ['.bam', '.bai']:
+        file_xml_dict = get_file_info(gnos_analysis.get('files').get('file'), file_type)
+        if file_xml_dict: 
+            file_xml = json.dumps(file_xml_dict, indent=4, sort_keys=True)
+            md5sum.append(hashlib.md5(file_xml).hexdigest())
+        else:
+            md5sum.append('missing')
+
+    if xml_dir: 
+        with open(xml_dir + '/' + fname +'_ordered_'+get_formal_repo_name(repo), 'w') as y:
+            y.write(effective_eq_xml)
+
+    return md5sum
+
+
+def get_gnos_repo(gnos_repo, source_gnos_repo, target_gnos_repo):
+    if not source_gnos_repo or not target_gnos_repo:
+        return gnos_repo
+
+    if source_gnos_repo in gnos_repo and target_gnos_repo in gnos_repo:
+        gnos_repo = [source_gnos_repo, target_gnos_repo]
+    else:
+        gnos_repo = []
+
+    return gnos_repo 
+
+
+def main(argv=None):
+
+    parser = ArgumentParser(description="Compare the effective xml md5sum among repos",
+             formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument("-m", "--metadata_dir", dest="metadata_dir",
+             help="Directory containing metadata manifest files", required=True)
+    parser.add_argument("-f", "--gnos_id_file", dest="fname",
+             help="Specify gnos_id file to process", required=True)
```

Comment:
What is this gnos_id_file? shouldn't the input be this report file: http://pancancer.info/gnos_metadata/latest/reports/QC_reports/specimens_with_mismatch_effective_xml_md5sum.txt? The task for this script is to add a few more columns to indicate where are the specific differences.